sycl/include/CL/sycl/ONEAPI/reduction.hpp

```diff
@@ -1050,6 +1093,465 @@ reduAuxCGFunc(handler &CGH, size_t NWorkItems, size_t MaxWGSize,
   return NWorkGroups;
 }
 
+/// For the given 'Reductions' types pack and indices enumerating only
+/// the reductions for which a local accessors are needed, this function creates
+/// those local accessors and returns a tuple consisting of them.
+template <typename... Reductions, size_t... Is>
+std::tuple<typename Reductions::local_accessor_type...>
+createReduLocalAccs(size_t Size, handler &CGH, std::index_sequence<Is...>) {
+  return {Reductions::getReadWriteLocalAcc(Size, CGH)...};
+}
```

Comment:
This is a good catch. This code works correctly now because local accessors are created for ALL reductions passed to here, i.e. the same tree-reduction algorithm is used for all.
Later (when I add optimized code-gen for reductions with fast_reduce and/or fast_atomic_fetch), I would discover this error here and fix it.
I am going to go ahead and fix it now.

Reply:
Fixed in 94e1be0649f968184de167880387b0dde63782c5